src/tools/src/bin/wal-reader.rs

```diff
@@ -0,0 +1,165 @@
+
+use std::{
+    fs::{self},
+    sync::Arc,
+};
+
+use clap::Parser;
+use runtime::Builder;
+use tabled::{Table, Tabled};
+use wal::local_storage_impl::segment::{Region, RegionManager, SegmentTableMeta};
+
+#[derive(Parser, Debug)]
+#[clap(author, version, about = "A command line tool to read and display WAL (Write-Ahead Log) segment metadata", long_about = None)]
+struct Args {
+    /// Type of WAL storage
+    #[clap(short, long, default_value = "local")]
+    r#type: String,
+
+    /// Data directory path
+    #[clap(short, long, default_value = "/tmp/horaedb/wal")]
+    data_dir: String,
+
+    /// Region id
+    #[clap(short = 'r', long, default_value = None)]
+    region_id: Option<u64>,
+
+    /// Segment id
+    #[clap(short = 's', long, default_value = None)]
+    segment_id: Option<u64>,
+
+    /// Table id
+    #[clap(long, default_value = None)]
+    table_id: Option<u64>,
+
+    /// Segment size in bytes
+    #[clap(short = 'z', long, default_value = "67108864")] // 64 * 1024 * 1024
+    segment_size: usize,
+}
+
+#[derive(Tabled)]
+struct SegmentInfo {
+    segment_id: u64,
+    min_seq: u64,
+    max_seq: u64,
+    version: u8,
+    current_size: usize,
+    segment_size: usize,
+    number_of_records: usize,
+}
+
+#[derive(Tabled)]
+struct TableInfo {
+    table_id: u64,
+    min_seq: u64,
+    max_seq: u64,
+}
+
+impl SegmentInfo {
+    fn load(stm: &SegmentTableMeta) -> Self {
+        Self {
+            segment_id: stm.id,
+            min_seq: stm.min_seq,
+            max_seq: stm.max_seq,
+            version: stm.version,
+            current_size: stm.current_size,
+            segment_size: stm.segment_size,
+            number_of_records: stm.number_of_records,
+        }
+    }
+}
+
+impl TableInfo {
+    fn load(stm: &SegmentTableMeta, table_id: &Option<u64>) -> Vec<TableInfo> {
+        let mut datas = Vec::new();
+        for (t_id, (min_seq, max_seq)) in stm.tables.iter() {
+            if table_id.is_some() && table_id.unwrap() != *t_id {
+                continue;
+            }
+            datas.push(TableInfo {
+                table_id: *t_id,
+                min_seq: *min_seq,
+                max_seq: *max_seq,
+            });
+        }
+        datas
+    }
+}
+
+fn region_meta_dump(region: Arc<Region>, table_id: &Option<u64>) {
+    let segments = region.meta();
+    for stm in segments.iter() {
+        println!("{}", "-".repeat(94));
+        let pretty_segment = Table::new([SegmentInfo::load(stm)]);
+        println!("{}", pretty_segment);
+        let pretty_table = Table::new(TableInfo::load(stm, table_id));
+        println!("{}", pretty_table);
+    }
+}
+
+fn pretty_error_then_exit(err_msg: &str) {
+    eprintln!("\x1b[31m{}\x1b[0m", err_msg);
+    std::process::exit(1);
+}
+
+fn main() {
+    let args = Args::parse();
+    println!("WAL type: {}", args.r#type);
+    println!("Data directory: {}", args.data_dir);
+
+    if args.r#type.to_lowercase() != "local" {
```

Comment:
I think we can remove `type` arg, we could infer wal type based on file name.

Users of this tool don't care about wal type.